### PR TITLE
Added GetLatestQuote (/intraday/latest)

### PR DESCRIPTION
**Pull request by Jenscaasen**

### CODE REVIEW

**Comment by orshe4** [comment]
Please fix conflicts so I could merge

**Comment by Jenscaasen** [comment]
Sorry for the conflicts, i am completely new to open source and github and pull requests. Should be solved now.

File: Marketstack/Services/MarketstackService.cs [hunk 1]
```diff
@@ -57,12 +58,21 @@ public IAsyncEnumerable<StockBar> GetStockEodBars(string stockSymbol, DateTime f
             string dateFromStr = fromDate.ToString("yyyy-MM-dd");
             string dateToStr = toDate.ToString("yyyy-MM-dd");
             return _httpClient.GetAsync<StockBar>($"http://api.marketstack.com/v1/eod?symbols={stockSymbol}&date_from={dateFromStr}&date_to={dateToStr}", _options.ApiToken, _throttled);
-        }
+        }
+
+
+        public Task<StockBar> GetLatestQuote(string stockSymbol)
+        {
+            return _httpClient.GetSingleAsync<StockBar>($"http://api.marketstack.com/v1/intraday/latest?symbols={stockSymbol}", _options.ApiToken, _throttled);
```
Comment (orshe4) [inline Marketstack/Services/MarketstackService.cs:66]: Change it to:
`return _httpClient.GetAsync<StockBar>($"http://api.marketstack.com/v1/intraday/latest?symbols={stockSymbol}", _options.ApiToken, _throttled).FirstAsync();`

File: Marketstack/Services/HttpClientExtensions.cs [hunk 2]
```diff
@@ -29,6 +30,16 @@ public static async IAsyncEnumerable<T> GetAsync<T>(this HttpClient httpClient,
 
                 pageResponse = await throttled.Run(() => httpClient.GetNextPageResponse(builder, pageResponse));
             }
+        }
+
+        public static async Task<T> GetSingleAsync<T>(this HttpClient httpClient, string url, string apiToken, Throttled throttled)
```
Comment (orshe4) [inline Marketstack/Services/HttpClientExtensions.cs:35]: I think that this function is not necessary

File: Marketstack.Tests/MarketstackServiceTests.cs [hunk 3]
```diff
@@ -47,13 +47,21 @@ public async Task GetStockEodBars_ReturnsBars()
                 .ToListAsync();            
             Assert.NotEmpty(bars);
             Assert.True(bars.Count > 100);
-        }
-
+        }
+
+        [Fact]
+        public async Task GetLatestQuote_ReturnsBars()
```
Comment (orshe4) [inline Marketstack.Tests/MarketstackServiceTests.cs:53]: Rename to `GetLatestQuote_ReturnsBar`

File: Marketstack.Tests/MarketstackServiceTests.cs [hunk 4]
```diff
@@ -47,13 +47,21 @@ public async Task GetStockEodBars_ReturnsBars()
...
-        }
-
+        }
+
+        [Fact]
+        public async Task GetLatestQuote_ReturnsBars()
+        {
+            var appleSymbol = "AAPL";
+            var bar = await _marketstackService.GetLatestQuote(appleSymbol);
+            Assert.NotNull(bar);
```
Comment (orshe4) [inline Marketstack.Tests/MarketstackServiceTests.cs:57]: You can add some assertions to the bar date. Validate that it is the latest bar.

**Review by orshe4 — CHANGES_REQUESTED** [review]

File: Marketstack/Services/MarketstackService.cs (hunk 1 above)
Comment (orshe4) [reply Marketstack/Services/MarketstackService.cs:66]: Please use the following api endpoints:

`/tickers/[symbol]/eod/latest` | Obtain the latest end-of-day data for a given stock symbol. 

`/tickers/[symbol]/intraday/latest` | Obtain the latest intraday data for a given stock symbol

**Review by orshe4 — CHANGES_REQUESTED** [review]

**Comment by orshe4** [comment]
> Sorry for the conflicts. I am completely new to open source and GitHub and pull requests. Should be solved now.

I finished the review. Please review my comments. Thanks again for contributing to my repo.
After the merge, I will update the NuGet.

**Comment by orshe4** [comment]
Closed due to inactivity. In case you would like to continue with this merge, please pulled from the master and review my comments.

Please note the API changes in `IMarketstackService.cs`

**Pull request closed without merging**